Backend/chat/chatController.js

```diff
@@ -0,0 +1,41 @@
+const chatModel = require("./chatModel.js");
+const io = require("../socket");
+
+exports.getAllMessages = async (req, res) => {
+  const { order_id } = req.params;
+  try {
+    const allMyOrders = await chatModel.findAllByOrderId(order_id);
+    if (!allMyOrders) {
+      return res.status(400).json({
+        errorMessage: "Something went wrong with your chat request",
+      });
+    }
+    return res.status(200).json(allMyOrders);
+  } catch (error) {
+    return res.status(500).json({
+      errorMessage: error,
+    });
+  }
+};
+
+exports.createChat = async (req, res) => {
+  try {
+    const savedMessage = await chatModel.saveMessage(req.body);
```

Comment:
Check

Reply:
You changed it in the chatController, but it's not changed in other controllers. Please check this thing in all your code.

I actually think we should add an internal models flow

Reply:
I checked all controllers properly all fields are Destructuring from req.body. 

Chat and User model you can see it directly in controller files because they have few fields coming from frontend side so no need separate file.

But order and offer controller has many fields so all fields are kept in separate files you can get them in their helpers functions.

Reply:
Check